pkg/tasks/metricsserver.go

```diff
@@ -104,9 +105,34 @@ func (t *MetricsServerTask) create(ctx context.Context) error {
 		if err != nil {
 			return fmt.Errorf("reconciling MetricsServer Service failed: %w", err)
 		}
-	}
-	{
-		dep, err := t.factory.MetricsServerDeployment()
+
+		kscm, err := t.client.GetConfigmap(ctx, "openshift-config-managed", "kubelet-serving-ca")
+		if err != nil {
+			return fmt.Errorf("openshift-config-managed/kubelet-serving-ca: %w", err)
+		}
+
+		cacm, err := t.factory.PrometheusK8sKubeletServingCABundle(kscm.Data)
+		if err != nil {
+			return fmt.Errorf("initializing kubelet serving CA Bundle ConfigMap failed: %w", err)
+		}
```

Comment:
We could just use `PrometheusK8sKubeletServingCABundle` with empty data as we're only interested in namespace/name to avoid hardcoding them and to signal that we rely on that resource as well.

```suggestion

               // Only interested in its name and to declare dependency on the asset.
		cacm, err := t.factory.PrometheusK8sKubeletServingCABundle([]string{})
		if err != nil {
			return fmt.Errorf("initializing an empty kubelet serving CA Bundle ConfigMap failed: %w", err)
		}
		then WaitForConfigMap(cacm)
```